libraries/redcore/api/soap/helper.php

```diff
@@ -44,13 +44,227 @@ public static function createSoapFaultResponse($message, $faultCode = 'SOAP-ENV:
+					$field, $sequence, $typeSchema,
+					($elementName != '' ? $elementName : $typeName),
+					$validateOptional,
+					(isset($field['fields']) ? $field['fields'] : array())
+				);
+			}
+		}
+	}
+
+	/**
+	 * Returns output resources by filtering out _links and _messages
+	 *
+	 * @param   SimpleXMLElement  $xmlElement        Xml element
+	 * @param   string            $resourceSpecific  Optionally limits the results to a certain specific resource
+	 * @param   boolean           $namesOnly         Optionally create an array of names only
+	 *
+	 * @return  array
+	 */
+	public static function getOutputResources($xmlElement, $resourceSpecific = '', $namesOnly = false)
+	{
+		$outputResources = array();
+
+		if (isset($xmlElement->resources->resource))
+		{
+			foreach ($xmlElement->resources->resource as $resource)
+			{
+				$displayGroup = RApiHalHelper::attributeToString($resource, 'displayGroup');
+
+				switch ($displayGroup)
+				{
+					case '_links':
+					case '_messages':
+						break;
+
+					default:
+						if (($resourceSpecific != '' && RApiHalHelper::attributeToString($resource, 'resourceSpecific') == $resourceSpecific)
+							|| $resourceSpecific == '')
+						{
+							if ($namesOnly)
+							{
+								$outputResources[] = RApiHalHelper::attributeToString($resource, 'displayName');
+							}
+							else
+							{
+								$resource->addAttribute('name', $resource['displayName']);
+								$outputResources[] = $resource;
+							}
+						}
+				}
+			}
+		}
+
+		return $outputResources;
+	}
+
+	/**
+	 * Returns the resoult resource from a certain operation
+	 *
+	 * @param   SimpleXMLElement  $xmlElement  Xml element
+	 *
+	 * @return  array
+	 */
+	public static function getResultResource($xmlElement)
+	{
+		if (isset($xmlElement->resources->resource))
+		{
+			foreach ($xmlElement->resources->resource as $resource)
+			{
+				$displayName = RApiHalHelper::attributeToString($resource, 'displayName');
+				$resourceSpecific = RApiHalHelper::attributeToString($resource, 'resourceSpecific');
+
+				if ($displayName == 'result' && $resourceSpecific == 'rcwsGlobal')
+				{
+					$resource->addAttribute('name', $resource['displayName']);
+
+					return $resource;
+				}
+			}
+		}
+
+		$resource = new SimpleXMLElement('<resource name="result" displayName="result" transform="array" fieldFormat="{result}" />');
+
+		return $resource;
+	}
+
+	/**
+	 * Method to determine the wsdl file name
+	 *
+	 * @param   string  $client          Client
+	 * @param   string  $webserviceName  Name of the webservice
+	 * @param   string  $version         Suffixes to the file name (ex. 1.0.0)
+	 * @param   string  $extension       Extension of the file to search
+	 * @param   string  $path            Path to webservice files
+	 *
+	 * @return  string  The full path to the api file
+	 *
+	 * @since   1.4
+	 */
+	public static function getWebserviceFile($client, $webserviceName, $version = '', $extension = 'xml', $path = '')
```

Comment:
It has a totally different behavior and it was the reason why wsdl wasn't actually cached.
The one in hal checks for existing files, this one only generates the name.  The other one is used to check if the file exists with different names.